core/modules/node/views/node.views.inc

```diff
@@ -734,11 +733,18 @@ function node_row_node_view_preprocess_node(&$variables) {
  * Implements hook_views_query_substitutions().
  */
 function node_views_query_substitutions() {
-  return array(
+  $substitutions = array(
     '***ADMINISTER_NODES***' => intval(user_access('administer nodes')),
     '***VIEW_OWN_UNPUBLISHED_NODES***' => intval(user_access('view own unpublished content')),
+    '***VIEW_ANY_UNPUBLISHED_NODES***' => intval(user_access('view any unpublished content')),
     '***BYPASS_NODE_ACCESS***' =>  intval(user_access('bypass node access')),
   );
+
+  foreach (node_type_get_types() as $type => $name) {
+    $substitutions["***VIEW_UNPUBLISHED_$type***"] = intval(user_access('view any unpublished ' . $type . ' content'));
+  }
```

Comment:
If we don't provide these permissions in hook_permission() we shouldn't bother with Views substitutions at this point.